fix(blog): blog tag audit and overhaul

gillkyle opened the pull request:
<!--
  Have any questions? Check out the contributing docs at https://gatsby.dev/contribute, or
  ask in this Pull Request and a Gatsby maintainer will be happy to help :)
-->

## Description

<!-- Write a brief description of the changes introduced by this PR -->

Me and @calcsam paired for a day on auditing the tags we have on the blog to cut down on topics that are too specific or redundant. 

The audit of tags happened on this sheet: https://docs.google.com/spreadsheets/d/1HwDkk55h85dLWJTPHk_3GvGfJmqqL_iunlOYPjdGbcs/edit#gid=0

We pulled from key marketing messages that have been outlined in narrowing the tags, the goal was to remove 1/3 of the existing tags but we ended up removing or renaming nearly 50% (**from 165 tags to 81**)

## Screenshots

A popular tags section was added to highlight some commonly used tags:

<details>
<summary>Tags page (BEFORE)</summary>

![image](https://user-images.githubusercontent.com/21114044/61649807-66f21800-ac6f-11e9-97f8-f1aa42167d7c.png)

</details>

<details>
<summary>Tags page (AFTER)</summary>

![image](https://user-images.githubusercontent.com/21114044/61650156-4ffff580-ac70-11e9-9a5a-7bf9c848ce6e.png)

</details>

And the tag details pages were updated to include a link back to docs for tags that have docs associated with them:

<details>
<summary>Tag details</summary>

![image](https://user-images.githubusercontent.com/21114044/61650332-b258f600-ac70-11e9-9e50-67f65e6ff1d2.png)


</details>

KyleAMathews commented:
This is 🔥 nice work!

Did you also add a link to the tags page from the blogs home page? IMO there should be one at the top.

gillkyle commented:
We didn't move it no, think it should be duplicated at top and bottom? Or just at the top?

KyleAMathews commented:
Ah I don't remember seeing the "view tags" button at the bottom — that's perhaps fine.

It could make sense putting the tags page in a more prominent place now that it's more useful & our list of blog posts keep getting bigger. I'm frequently sending people links to blog posts which seems to indicate a need for easier browsing of blog posts.

marcysutton commented on docs/blog/2018-09-27-reach-router/index.md:
```diff
@@ -25,25 +25,51 @@ When a user navigates between traditional server-rendered pages, the page is ful
+  src="https://www.youtube.com/embed/NKTdNv8JpuM?rel=0"
+  frameborder="0"
+  allow="autoplay; encrypted-media"
+  allowfullscreen
```
```suggestion
  allowfullscreen
  title="YouTube: (React) Disqus router">
```

pieh commented:
Nice!

Not for this PR - is it worth trying to enforce whitelisted tags so those don't "swell" in few months again?

@freiksenet recently added this to starter/site showcase in peril ( https://github.com/gatsbyjs/gatsby/blob/master/docs/categories.yml ) - might use this for blog posts too

marcysutton commented on docs/blog/2018-10-09-hacktoberfest-kickoff/index.md:
```diff
@@ -2,7 +2,7 @@
 title: "Celebrate Hacktoberfest with Gatsby!"
 date: 2018-10-09
 author: "Amberley Romo"
-tags: ["hacktoberfest", "open source"]
```
Open source makes sense to keep as a tag, in my opinion, to help us explore the line between OSS contributions and support of funded products. People obviously care a lot about that distinction and we need to do a better job as a company of explaining our philosophy on it–so tagging content as "open source" provides a low-effort way for us to do that. I also think it's confusing to tag this as "gatsby-inc" as it blurs the line too much.

marcysutton commented on docs/blog/2018-10-12-uptick-docs-contributions-hacktoberfest/index.md:
```diff
@@ -2,7 +2,7 @@
 date: 2018-10-12
 author: Shannon Soper
-tags: ["documentation", "hacktoberfest"]
+tags: ["gatsby-docs", "hacktoberfest"]
```
```suggestion
tags: ["gatsby-docs", "hacktoberfest", "open-source"]
```

marcysutton commented on docs/blog/2018-11-01-hacktoberfest-wrapup/index.md:
```diff
@@ -2,7 +2,7 @@
 date: 2018-11-01
 author: "Amberley Romo"
-tags: ["hacktoberfest", "open source"]
+tags: ["hacktoberfest", "gatsby-inc"]
```
```suggestion
tags: ["hacktoberfest", "open-source"]
```
See above reasoning re: open source vs. company

marcysutton commented on docs/blog/2018-12-19-kentico-cloud-and-gatsby-take-you-beyond-static-websites/index.md:
```diff
@@ -4,7 +4,7 @@ date: 2019-01-07
 image: "images/teaser.png"
 imageTitle: Kentico Cloud and Gatsby
-tags: ["react", "kentico cloud", "graphql"]
+tags: ["react", "graphql"]
```
```suggestion
tags: ["react", "graphql", "content-mesh"]
```
Would "content-mesh" make sense to add in place of "kentico cloud" or even "sourcing"?

marcysutton commented on docs/blog/2018-12-31-how-we-do-high-impact-ux-research/index.md:
```diff
@@ -3,7 +3,7 @@ title: How Gatsby does high-impact, low-effort UX research (and you can too!)
 date: 2018-12-31
 tags:
-  - ux
+  - gatsby-user-testing
```
```suggestion
  - ux
  - gatsby-user-testing
```
I'd recommend keeping this

marcysutton commented on docs/blog/2018-2-3-sites-with-headless-cms/index.md:
```diff
@@ -2,7 +2,7 @@
 date: 2018-02-03
 author: "Shannon Soper"
-tags: ["headless-cms", "contentful", "graphcms", "cloudcms"]
+tags: ["headless-cms", "contentful"]
```
```suggestion
tags: ["headless-cms", "contentful", "content-mesh"]
```

marcysutton commented on docs/blog/2018-2-7-jam-out-your-blog/index.md:
```diff
@@ -73,7 +73,7 @@ Prose was originally created for editing markdown files on GitHub by the people
 
 - Simple Content Authoring Environment
-- Designed for CMS-Free Websites
+- designed for CMS-Free Websites
```
This seems like it should remain upper-cased since all the other items are upper-cased
```suggestion
- Designed for CMS-Free Websites
```

marcysutton commented on docs/blog/2019-01-29-gatsby-eng-hiring-philosophy/index.md:
```diff
@@ -130,7 +131,7 @@ In general, it’s a huge bonus when engineers have skills or experience such as
 - Product thinking
 
-- Design skills
+- design skills
```
I think this one should also stay upper-cased
```suggestion
- Design skills
```

marcysutton commented on docs/blog/2019-01-31-why-themes/index.md:
```diff
@@ -45,7 +45,7 @@ Plugins have been a very successful lower-level abstraction for Gatsby. Themes i
 - OSS/Commercial entrepreneurs who want to create web products. Themes will enable them to assemble a product they can share or sell with others.
 
-- Designers who want to create unique/beautiful website designs to share with others
+- designers who want to create unique/beautiful website designs to share with others
```
```suggestion
- Designers who want to create unique/beautiful website designs to share with others
```

marcysutton commented on docs/blog/2019-03-05-dot-org-messaging-survey/index.md:
```diff
@@ -2,7 +2,7 @@
 date: 2019-03-05
 author: Shannon Soper
-tags: ["ux", "design"]
+tags: ["gatsby-user-testing", "cutting-edge-experiences"]
```
```suggestion
tags: ["ux", "gatsby-user-testing", "cutting-edge-experiences"]
```

marcysutton commented on docs/blog/2019-03-11-dot-org-prototypes/index.md:
```diff
@@ -2,7 +2,7 @@
 date: 2019-03-11
 author: Shannon Soper
-tags: ["ux", "design"]
+tags: ["gatsby-user-testing", "cutting-edge-experiences"]
```
```suggestion
tags: ["ux", "gatsby-user-testing", "cutting-edge-experiences"]
```

marcysutton commented on docs/blog/2019-03-22-introducing-gatsby-preview-beta/index.md:
```diff
@@ -4,9 +4,8 @@ date: 2019-03-22
-  - cloud
-  - commercial
+  - gatsby-preview
+  - gatsby-cloud
```
```suggestion
  - gatsby-cloud
  - gatsby-inc
```

marcysutton commented on docs/blog/2019-06-08-delicious-simplicity-case-study-part-1/index.md:
```diff
@@ -3,8 +3,8 @@ title: "Delicious Simplicity ❤️ Gatsby: Building Happily Ever After"
 author: Michelle Gienow
 tags:
-  - case study
-  - developer experience
```
People talk about `developer-experience` as one of the things they like about Gatsby all the time–so that seems more intuitive to me as a tag than `building-sites-faster`. There are also parts of DX that aren't necessarily about "building sites faster", or "building faster sites" which is also what that brings to mind. Could we keep both tags?

gillkyle replied:
I think that's a good point, doesn't hurt to have both by any means.

marcysutton commented on docs/blog/2019-06-17-day-in-the-life-of-a-core-maintainer/index.md:
```diff
@@ -5,8 +5,7 @@ author: Dustin Schau
   - community
-  - open source
-  - contributor
+  - gatsby-inc
```
```suggestion
  - gatsby-inc
  - open-source
```
I'd recommend keeping this, as it's another area that shows how company and OSS overlap: Gatsby employs full-time maintainers to benefit the open source community

marcysutton commented on docs/blog/2019-06-19-open-beta-for-gatsby-preview-is-here/index.md:
```diff
@@ -4,8 +4,7 @@ date: 2019-06-19
 tags:
-  - commercial
-  - Gatsby Preview
+  - gatsby-preview
```
```suggestion
  - gatsby-preview
  - gatsby-inc
```

marcysutton commented on docs/docs/adding-search-with-algolia.md:
```diff
@@ -593,3 +593,5 @@ If everything works as expected, running `gatsby develop` should now give you so
 
 `youtube: VSkXyuXzwlc`
+
+You can also find stories of companies using Gatsby + Algolia together [in the Algolia section of the blog](/blog/algolia).
```
This page is a 404. Should it go to `/blog/tags/algolia` instead?

gillkyle replied:
```suggestion
You can also find stories of companies using Gatsby + Algolia together [in the Algolia section of the blog](/blog/tags/algolia).
```

marcysutton commented on docs/docs/gatsby-for-agencies.md:
```diff
@@ -5,3 +5,5 @@ title: Using Gatsby Inside Agencies
 
 <GuideList slug={props.slug} />
+
+The Gatsby blog has a [section on agencies](/blog/tags/agencies/) with additional stories and tips.
```
Nice. These links will be useful!

marcysutton commented on www/src/data/tags-docs.js:
```diff
@@ -0,0 +1,89 @@
+export const TAGS_AND_DOCS = new Map([
+  [`accessibility`, `/docs/making-your-site-accessible/`],
```
We should document this somewhere, since it wasn't immediately clear to me what it was for. I was also worried these hard-coded links would be brittle – particularly the headings (which perhaps should be pulled into their own docs if they're that important, e.g. `#client-only-routes--user-authentication`?). 

To document, we could add a mention of blog tags & docs on the Gatsby Style Guide and perhaps in https://www.gatsbyjs.org/contributing/blog-and-website-contributions/ and https://www.gatsbyjs.org/contributing/docs-contributions/

gillkyle replied:
I agree that the headings could be brittle and moving them into their own guides seems like a good solution. Do you think that falls into the scope of a separate PR though?

I just made some updates to comment and document how to add tags, and how the whitelist works.

marcysutton replied:
New guides can definitely go in separate PRs! It would be good to file issues for them, however. Here are the ones I think could go in separate docs; if you've got any time to file issues for them that would be amazing!

- client-only-routes--user-authentication
- construct-new-higher-level-web-building-blocks

I think that's it. I left comments on the other ones!

gillkyle replied:
I'll go in and add issues, went down a rabbit hole finding missing pages from the sidebar when I started though woops

KyleAMathews commented on docs/blog/2017-12-20-introducing-the-gatsby-ux-research-program/index.md:
```diff
@@ -2,7 +2,7 @@
 date: 2017-12-20
 author: "Shannon Soper"
-tags: ["ux"]
+tags: ["gatsby-user-testing"]
```
Yeah -- also user testing isn't a very good description as that's only a portion of what these posts are about. 

What's the rational for all the gatsby-* tags? They seem more cumbersome and won't make them more googable or scannable in the list of tags.

gillkyle replied:
Definitely open to suggestions like this of what to keep. The `gatsby-*` tags felt less ambiguous than a tag like `apps`, where `gatsby-for-apps` had more meaning.

KyleAMathews replied:
I think given it's on our blog the context is clear and doesn't need the extra words.

We could add descriptions for tag pages as well that make it clear what that tag is about e.g. "Gatsby is great for apps! These blog posts talk about this, etc. etc."

gillkyle commented:
@calcsam do you have an opinion on any of the suggested changes? Once everyone's weighed in I'll go through and clear up all the suggestions.

> Not for this PR - is it worth trying to enforce whitelisted tags so those don't "swell" in few months again?

This comment https://github.com/gatsbyjs/gatsby/pull/15979#discussion_r305969177 points out that this needs to be documented but the idea was have that map of tags as the source to pull from that we can add/remove from.

DSchau commented on docs/blog/2017-02-21-1-0-progress-update-where-came-from-where-going/index.md:
```diff
@@ -221,7 +221,7 @@ Plugins also drive the new GraphQL data processing layer.
 The final pieces for theme support are coming soon. Gatsby's theme system will
 have some unique properties. You'll be able to add multiple themes to a site. So
-you could start a site with a "blog" theme and later add on a "portfolio" theme.
+you could start a site with a "gatsby-for-blogs" theme and later add on a "portfolio" theme.
```
Bitten by find and replace, haha 🙃 

```suggestion
you could start a site with a "blog" theme and later add on a "portfolio" theme.
```

gillkyle commented:
Just pushed some updates:

- moved the "View All" button to the top of the page:

![image](https://user-images.githubusercontent.com/21114044/61833410-71f4a600-ae31-11e9-9055-87a9ed1cc2ef.png)

- went through and changed `gatsby-for-*` tags into just what the last word was: `gatsby-for-apps` -> `apps`

- added some comments and notes in the docs about the tag whitelist

- added some more tags to posts that were relevant

calcsam commented:
Thanks for the feedback @marcysutton @DSchau @KyleAMathews @pieh! This was a fun bit of spring cleaning. 

Re ergonomics of "enforcing" tags -- agree with @gillkyle on that.

marcysutton commented on docs/blog/2018-07-17-announcing-gatsby-preview/index.md:
```diff
@@ -2,7 +2,7 @@
 date: 2018-07-17
 author: "Sam Bhagwat"
-tags: ["contentful", "preview", "product"]
+tags: ["contentful", "gatsby-preview", "gatsby-cloud"]
```
```suggestion
tags: ["contentful", "gatsby-preview", "gatsby-cloud", "gatsby-inc"]
```

marcysutton commented on www/src/data/tags-docs.js:
```diff
@@ -0,0 +1,98 @@
+  [`apis`, ``],
+  [`apps`, ``],
+  [`authentication`, `/docs/building-a-site-with-authentication/`],
+  [`aws`, `/docs/deploying-to-s3-cloudfront/#getting-started---aws-cli`],
```
```suggestion
  [`aws`, `/docs/deploying-to-s3-cloudfront/`],
```
The paragraph referenced is at the beginning of the page, so I think it makes sense to link to the overall doc.

marcysutton commented on www/src/data/tags-docs.js:
```diff
@@ -0,0 +1,98 @@
+  [`drupal`, `/docs/sourcing-from-drupal/`],
+  [`ecommerce`, ``],
+  [`enterprise`, `/docs/gatsby-in-the-enterprise/`],
+  [`gatsby-apis`, `/docs/api-specification/#api`],
```
```suggestion
  [`gatsby-apis`, `/docs/api-reference/`],
```
Should we link to API reference instead of going to the middle of that page? Or just go to the broader API specification page?

gillkyle replied:
I think the broader page is a good call

lannonbr commented:
Thanks so much for this @gillkyle and @calcsam 🎉
